frontend/mulighetsrommet-veileder-flate/src/apps/modia/historikk/Statusbadge.tsx

```diff
@@ -1,32 +1,118 @@
-import classNames from "classnames";
-import { HistorikkForBrukerStatus } from "mulighetsrommet-api-client";
+import { AmtDeltakerStatus, ArenaDeltakerStatus } from "mulighetsrommet-api-client";
+import { Tag } from "@navikt/ds-react";
 import styles from "./Statusbadge.module.scss";
-import { Detail } from "@navikt/ds-react";
+import classNames from "classnames";
+
+type Variant =
+  | "warning"
+  | "warning-filled"
+  | "warning-moderate"
+  | "error"
+  | "error-filled"
+  | "error-moderate"
+  | "info"
+  | "info-filled"
+  | "info-moderate"
+  | "success"
+  | "success-filled"
+  | "success-moderate"
+  | "neutral"
+  | "neutral-filled"
+  | "neutral-moderate"
+  | "alt1"
+  | "alt1-filled"
+  | "alt1-moderate"
+  | "alt2"
+  | "alt2-filled"
+  | "alt2-moderate"
+  | "alt3"
+  | "alt3-filled"
+  | "alt3-moderate"
+  | "white"; // Ekstra variant for hvit bakgrunn
+
+export function StatusBadge({ status }: { status: AmtDeltakerStatus | ArenaDeltakerStatus }) {
+  const { variant, title } = variantAndTitle(status);
 
-export function StatusBadge({ status }: { status?: HistorikkForBrukerStatus }) {
   return (
-    <Detail
-      className={classNames(
-        styles.historikk_for_bruker_statusbadge,
-        styles[status as unknown as any],
-      )}
+    <Tag
+      size="small"
+      variant={variant === "white" ? "neutral" : variant}
+      className={classNames(variant === "white" && styles.white)}
```

Comment:
Du kan også bruke classNames slik:
```
classNames({[styles.white]: variant === "white})
```
Om det er så mye mer lesbart kan diskuteres, så du kan kjenne på det selv hva du går for.